godot-cell/tests/mock_blocking.rs

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! A mock implementation of our instance-binding pattern in pure rust for the blocking variant of GdCell.
+//!
+//! Used so we can run miri on this, which we cannot when we are running in itest against Godot.
```

Comment:
It would be nice if you could clearly separate
- mock (i.e. the parts that mirror the gdext API)
- additional code that uses the mocks and is used for testing
- actual tests

You could use the `// ----` style separators that you find in many parts of the library, just copy them :slightly_smiling_face: